### PR TITLE
Migrate some IGNITION_* macros

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Part of https://github.com/gazebo-tooling/release-tools/issues/698

## Summary

The `IGNITION_*` macros from the autogenerated `Export.hh` have been migrated to support `GZ_*` as well as of https://github.com/gazebosim/gz-cmake/pull/249, so the `GZ_UTILS_VISIBLE` macro is now being used (https://github.com/gazebosim/gz-utils/commit/97a1cfa12fc7e2ebf9c0c4142736531779929706). This also tick-tocks most of the macros in `config.hh.in` (notably not the `*_BUILD_TYPE_*` macros, which are problematic) and migrates to `GZ_UTILS_VERSION_NAMESPACE` where relevant.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/gz/utils/config.hh.in [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+#define GZ_UTILS_VERSION_FULL "${PROJECT_VERSION_FULL}"
+#define IGNITION_UTILS_VERSION GZ_UTILS_VERSION
+#define IGNITION_UTILS_VERSION_FULL GZ_UTILS_VERSION_FULL
 
-#define IGNITION_UTILS_VERSION_NAMESPACE v${PROJECT_VERSION_MAJOR}
+#define GZ_UTILS_VERSION_NAMESPACE v${PROJECT_VERSION_MAJOR}
+#define IGNITION_UTILS_VERSION_NAMESPACE GZ_UTILS_VERSION_NAMESPACE
 
-#define IGNITION_UTILS_VERSION_HEADER "Ignition ${IGN_DESIGNATION}, version ${PROJECT_VERSION_FULL}\nCopyright (C) 2020 Open Source Robotics Foundation.\nReleased under the Apache 2.0 License.\n\n"
+#define GZ_UTILS_VERSION_HEADER "Ignition ${IGN_DESIGNATION}, version ${PROJECT_VERSION_FULL}\nCopyright (C) 2020 Open Source Robotics Foundation.\nReleased under the Apache 2.0 License.\n\n"
```
Comment (methylDragon) [inline include/gz/utils/config.hh.in:19]: Should the `Ignition` change to `Gz`?

Comment (chapulina) [reply include/gz/utils/config.hh.in:19]: I updated that `Ignition` has part of another PR series, see https://github.com/gazebosim/gz-gui/pull/398/files#diff-eb6b9b2cedd00e7c9a1bfa077a11ad2eefed50f2925c9f05c83cf0f3fdae01bd. I don't really mind where we do it, as long as we get rid of all `Ignition` occurrences by the end.

**Review by chapulina — APPROVED** [review]
LGTM, I guess there's no way to trigger a warning when the `IGNITION_` versions are used, right? It's more of a slient-tick and then tock, but I guess that's what we can do.

I believe this is backportable to Citadel and Fortress if we want to reduce the diff between the branches. I don't think we should do that now, we may have time for it after feature freeze.

**Comment by scpeters** [comment]
> LGTM, I guess there's no way to trigger a warning when the `IGNITION_` versions are used, right? It's more of a slient-tick and then tock, but I guess that's what we can do.

once you mentioned that, it occurred to me that we could move the `IGNITION_*` macros to `ignition/utils/config.hh`, so I did so in https://github.com/gazebosim/gz-utils/pull/62/commits/06f7b943b063181dd5603e45b1d8040c16b0db52

how does that look?

**Review by chapulina — APPROVED** [review]

**Pull request merged**